```
make test
CLICKHOUSE_VERSION=latest go test -v -race `go list ./... | grep -v ./internal/platform/test`
?   	github.com/ClickHouse/clickhouse-diagnostics	[no test files]
?   	github.com/ClickHouse/clickhouse-diagnostics/cmd	[no test files]
=== RUN   TestNewParamMap
=== RUN   TestNewParamMap/test_param_map_correctly_converts_types
--- PASS: TestNewParamMap (0.00s)
    --- PASS: TestNewParamMap/test_param_map_correctly_converts_types (0.00s)
=== RUN   TestConvertParamsToConfig
=== RUN   TestConvertParamsToConfig/test_we_can_convert_a_param_map_back_to_a_config
--- PASS: TestConvertParamsToConfig (0.00s)
    --- PASS: TestConvertParamsToConfig/test_we_can_convert_a_param_map_back_to_a_config (0.00s)
=== RUN   TestAddParamMapToCmd
=== RUN   TestAddParamMapToCmd/test_we_can_add_hidden_params_to_a_command
--- PASS: TestAddParamMapToCmd (0.00s)
    --- PASS: TestAddParamMapToCmd/test_we_can_add_hidden_params_to_a_command (0.00s)
=== RUN   TestStringOptionsVar
=== RUN   TestStringOptionsVar/test_we_can_set
=== RUN   TestStringOptionsVar/test_set_invalid
--- PASS: TestStringOptionsVar (0.00s)
    --- PASS: TestStringOptionsVar/test_we_can_set (0.00s)
    --- PASS: TestStringOptionsVar/test_set_invalid (0.00s)
=== RUN   TestStringSliceOptionsVar
=== RUN   TestStringSliceOptionsVar/test_we_can_set
=== RUN   TestStringSliceOptionsVar/test_set_invalid
--- PASS: TestStringSliceOptionsVar (0.00s)
    --- PASS: TestStringSliceOptionsVar/test_we_can_set (0.00s)
    --- PASS: TestStringSliceOptionsVar/test_set_invalid (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/cmd/params	(cached)
2022/06/20 21:18:18 Starting container id: fc5a06b4ce82 image: testcontainers/ryuk:0.3.3
2022/06/20 21:18:18 Waiting for container id fc5a06b4ce82 image: testcontainers/ryuk:0.3.3
2022/06/20 21:18:18 Container is ready id: fc5a06b4ce82 image: testcontainers/ryuk:0.3.3
2022/06/20 21:18:18 Starting container id: 5f838289cff9 image: clickhouse/clickhouse-server:latest
2022/06/20 21:18:19 Waiting for container id 5f838289cff9 image: clickhouse/clickhouse-server:latest
2022/06/20 21:18:20 Container is ready id: 5f838289cff9 image: clickhouse/clickhouse-server:latest
=== RUN   TestCapture
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"connection established"}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"initializing logs collector"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector logs: directory /var/log/clickhouse-server does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector logs: directory /etc/clickhouse-server does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector logs: directory /etc/clickhouse-server does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector logs: directory /etc/clickhouse-server/ does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector logs: directory /var/lib/clickhouse/preprocessed_configs does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector logs: directory /etc/clickhouse-server does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector logs: directory /etc/clickhouse-server does not exist"}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"initializing summary collector"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query top_queries_by_memory: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query top_queries_by_memory: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query top_queries_by_memory: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query crash_log: code: 60, message: Table system.crash_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query stack_traces: code: 446, message: default: Introspection functions are disabled, because setting 'allow_introspection_functions' is set to 0: While processing concat('\\n', arrayStringConcat(arrayMap((x, y) -> concat(x, ': ', y), arrayMap(x -> addressToLine(x), trace), arrayMap(x -> demangle(addressToSymbol(x)), trace)), '\\n')) AS trace"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query top_queries_by_duration: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query top_queries_by_duration: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query top_queries_by_duration: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query failed_queries: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query failed_queries: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector summary: Unable to execute query failed_queries: code: 60, message: Table system.query_log doesn't exist"}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"initializing system collector"}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"initializing config collector"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector config: directory /etc/clickhouse-server does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector config: directory /etc/clickhouse-server/ does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector config: directory /var/lib/clickhouse/preprocessed_configs does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector config: directory /etc/clickhouse-server does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector config: directory /etc/clickhouse-server does not exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector config: directory /etc/clickhouse-server does not exist"}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"initializing db_logs collector"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector db_logs: Unable to collect query_log: code: 60, message: Table system.query_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector db_logs: Unable to collect query_thread_log: code: 60, message: Table system.query_thread_log doesn't exist"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector db_logs: Unable to collect text_log: code: 60, message: Table system.text_log doesn't exist"}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"initializing system_db collector"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector system_db: Unable to collect certificates: code: 1000, message: SSL Exception: Configuration error: no certificate file has been specified"}
{"level":"warn","time":"2022-06-20T21:18:20+02:00","message":"Failure to collect frame in collector system_db: Unable to collect merge_tree_metadata_cache: code: 36, message: SELECT from system.merge_tree_metadata_cache table must contain condition like key = 'key' or key LIKE 'prefix%' in WHERE clause."}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"collectors initialized"}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"attempting to export bundle using simple output..."}
{"level":"info","time":"2022-06-20T21:18:20+02:00","message":"creating bundle in /tmp/TestCapture3797413038/001/random/1655752700909"}
{"level":"info","time":"2022-06-20T21:18:21+02:00","message":"bundle created"}
{"level":"info","time":"2022-06-20T21:18:21+02:00","message":"bundle export complete"}
--- PASS: TestCapture (0.54s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal	(cached)
=== RUN   TestGetCollectorNames
=== RUN   TestGetCollectorNames/can_get_all_collector_names
=== RUN   TestGetCollectorNames/can_get_default_collector_names
--- PASS: TestGetCollectorNames (0.00s)
    --- PASS: TestGetCollectorNames/can_get_all_collector_names (0.00s)
    --- PASS: TestGetCollectorNames/can_get_default_collector_names (0.00s)
=== RUN   TestGetCollectorByName
=== RUN   TestGetCollectorByName/can_get_collector_by_name
=== RUN   TestGetCollectorByName/fails_on_non_existing_collector
--- PASS: TestGetCollectorByName (0.00s)
    --- PASS: TestGetCollectorByName/can_get_collector_by_name (0.00s)
    --- PASS: TestGetCollectorByName/fails_on_non_existing_collector (0.00s)
=== RUN   TestBuildConfigurationOptions
=== RUN   TestBuildConfigurationOptions/can_get_all_collector_configurations
--- PASS: TestBuildConfigurationOptions (0.00s)
    --- PASS: TestBuildConfigurationOptions/can_get_all_collector_configurations (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/collectors	(cached)
=== RUN   TestConfigConfiguration
=== RUN   TestConfigConfiguration/correct_configuration_is_returned_for_config_collector
--- PASS: TestConfigConfiguration (0.00s)
    --- PASS: TestConfigConfiguration/correct_configuration_is_returned_for_config_collector (0.00s)
=== RUN   TestConfigCollect
=== RUN   TestConfigCollect/test_default_file_collector_configuration
=== RUN   TestConfigCollect/test_configuration_when_specified
--- PASS: TestConfigCollect (0.01s)
    --- PASS: TestConfigCollect/test_default_file_collector_configuration (0.01s)
    --- PASS: TestConfigCollect/test_configuration_when_specified (0.00s)
=== RUN   TestDbLogsConfiguration
=== RUN   TestDbLogsConfiguration/correct_configuration_is_returned_for_summary_collector
--- PASS: TestDbLogsConfiguration (0.00s)
    --- PASS: TestDbLogsConfiguration/correct_configuration_is_returned_for_summary_collector (0.00s)
=== RUN   TestDbLogsCollect
=== RUN   TestDbLogsCollect/test_default_db_logs_collection
=== RUN   TestDbLogsCollect/test_db_logs_collection_with_limit
--- PASS: TestDbLogsCollect (0.00s)
    --- PASS: TestDbLogsCollect/test_default_db_logs_collection (0.00s)
    --- PASS: TestDbLogsCollect/test_db_logs_collection_with_limit (0.00s)
=== RUN   TestLogsConfiguration
=== RUN   TestLogsConfiguration/correct_configuration_is_returned_for_logs_collector
--- PASS: TestLogsConfiguration (0.00s)
    --- PASS: TestLogsConfiguration/correct_configuration_is_returned_for_logs_collector (0.00s)
=== RUN   TestLogsCollect
=== RUN   TestLogsCollect/test_default_logs_collection
=== RUN   TestLogsCollect/test_logs_collection_when_directory_is_specified
    logs_test.go:145: 
        	Error Trace:	logs_test.go:145
        	            				logs_test.go:75
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 0
        	Test:       	TestLogsCollect/test_logs_collection_when_directory_is_specified
=== RUN   TestLogsCollect/test_logs_collection_of_archives
    logs_test.go:145: 
        	Error Trace:	logs_test.go:145
        	            				logs_test.go:97
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 1
        	Test:       	TestLogsCollect/test_logs_collection_of_archives
=== RUN   TestLogsCollect/test_when_directory_does_not_exist
--- FAIL: TestLogsCollect (0.01s)
    --- PASS: TestLogsCollect/test_default_logs_collection (0.01s)
    --- FAIL: TestLogsCollect/test_logs_collection_when_directory_is_specified (0.00s)
    --- FAIL: TestLogsCollect/test_logs_collection_of_archives (0.00s)
    --- PASS: TestLogsCollect/test_when_directory_does_not_exist (0.00s)
=== RUN   TestSummaryConfiguration
=== RUN   TestSummaryConfiguration/correct_configuration_is_returned_for_summary_collector
--- PASS: TestSummaryConfiguration (0.00s)
    --- PASS: TestSummaryConfiguration/correct_configuration_is_returned_for_summary_collector (0.00s)
=== RUN   TestSummaryCollection
=== RUN   TestSummaryCollection/test_default_summary_collection
=== RUN   TestSummaryCollection/test_summary_collection_with_limit
--- PASS: TestSummaryCollection (0.02s)
    --- PASS: TestSummaryCollection/test_default_summary_collection (0.01s)
    --- PASS: TestSummaryCollection/test_summary_collection_with_limit (0.01s)
=== RUN   TestSystemConfiguration
=== RUN   TestSystemConfiguration/correct_configuration_is_returned_for_system_db_collector
--- PASS: TestSystemConfiguration (0.00s)
    --- PASS: TestSystemConfiguration/correct_configuration_is_returned_for_system_db_collector (0.00s)
=== RUN   TestSystemDbCollect
=== RUN   TestSystemDbCollect/test_default_system_db_collection
=== RUN   TestSystemDbCollect/test_when_we_pass_an_includes
=== RUN   TestSystemDbCollect/test_when_we_pass_an_excludes
=== RUN   TestSystemDbCollect/test_when_we_pass_an_invalid_includes
=== RUN   TestSystemDbCollect/test_when_we_use_a_table_with_excluded_fields
=== RUN   TestSystemDbCollect/test_with_a_low_row_limit
=== RUN   TestSystemDbCollect/test_with_a_negative_low_row_limit
=== RUN   TestSystemDbCollect/test_that_includes_overrides_excludes
=== RUN   TestSystemDbCollect/test_banned
=== RUN   TestSystemDbCollect/test_banned_unless_included
=== RUN   TestSystemDbCollect/tables_are_ordered_if_configured
--- PASS: TestSystemDbCollect (0.00s)
    --- PASS: TestSystemDbCollect/test_default_system_db_collection (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_pass_an_includes (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_pass_an_excludes (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_pass_an_invalid_includes (0.00s)
    --- PASS: TestSystemDbCollect/test_when_we_use_a_table_with_excluded_fields (0.00s)
    --- PASS: TestSystemDbCollect/test_with_a_low_row_limit (0.00s)
    --- PASS: TestSystemDbCollect/test_with_a_negative_low_row_limit (0.00s)
    --- PASS: TestSystemDbCollect/test_that_includes_overrides_excludes (0.00s)
    --- PASS: TestSystemDbCollect/test_banned (0.00s)
    --- PASS: TestSystemDbCollect/test_banned_unless_included (0.00s)
    --- PASS: TestSystemDbCollect/tables_are_ordered_if_configured (0.00s)
=== RUN   TestZookeeperConfiguration
=== RUN   TestZookeeperConfiguration/correct_configuration_is_returned_for_system_zookeeper_collector
--- PASS: TestZookeeperConfiguration (0.00s)
    --- PASS: TestZookeeperConfiguration/correct_configuration_is_returned_for_system_zookeeper_collector (0.00s)
=== RUN   TestZookeeperCollect
=== RUN   TestZookeeperCollect/test_default_zookeeper_collection
--- PASS: TestZookeeperCollect (0.00s)
    --- PASS: TestZookeeperCollect/test_default_zookeeper_collection (0.00s)
FAIL
FAIL	github.com/ClickHouse/clickhouse-diagnostics/internal/collectors/clickhouse	0.051s
=== RUN   TestCommandConfiguration
=== RUN   TestCommandConfiguration/correct_configuration_is_returned_for_file_collector
--- PASS: TestCommandConfiguration (0.00s)
    --- PASS: TestCommandConfiguration/correct_configuration_is_returned_for_file_collector (0.00s)
=== RUN   TestCommandCollect
=== RUN   TestCommandCollect/test_simple_command_with_args
[ls -l ../../../testdata total 0
drwxr-xr-x 6 felixoid felixoid 62 Jun 19 01:43 configs
drwxr-xr-x 2 felixoid felixoid 41 Jun 19 01:43 docker
drwxr-xr-x 3 felixoid felixoid 17 Jun 19 01:43 logs
  ]
=== RUN   TestCommandCollect/test_empty_command
=== RUN   TestCommandCollect/test_invalid_command
--- PASS: TestCommandCollect (0.01s)
    --- PASS: TestCommandCollect/test_simple_command_with_args (0.01s)
    --- PASS: TestCommandCollect/test_empty_command (0.00s)
    --- PASS: TestCommandCollect/test_invalid_command (0.00s)
=== RUN   TestFileConfiguration
=== RUN   TestFileConfiguration/correct_configuration_is_returned_for_file_collector
--- PASS: TestFileConfiguration (0.00s)
    --- PASS: TestFileConfiguration/correct_configuration_is_returned_for_file_collector (0.00s)
=== RUN   TestFileCollect
=== RUN   TestFileCollect/test_filter_patterns_work
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/configs/include/xml/server-include.xml"}
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/configs/include/xml/user-include.xml"}
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/configs/xml/config.xml"}
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/configs/xml/users.xml"}
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/configs/xml/users.d/default-password.xml"}
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/configs/yandex_xml/config.xml"}
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/docker/admin.xml"}
{"level":"debug","time":"2022-06-20T21:18:17+02:00","message":"Collecting file ../../../testdata/docker/custom.xml"}
=== RUN   TestFileCollect/invalid_file_patterns_are_detected
=== RUN   TestFileCollect/check_empty_matches_are_reported
--- PASS: TestFileCollect (0.00s)
    --- PASS: TestFileCollect/test_filter_patterns_work (0.00s)
    --- PASS: TestFileCollect/invalid_file_patterns_are_detected (0.00s)
    --- PASS: TestFileCollect/check_empty_matches_are_reported (0.00s)
=== RUN   TestSystemConfiguration
=== RUN   TestSystemConfiguration/correct_configuration_is_returned_for_system_collector
--- PASS: TestSystemConfiguration (0.00s)
    --- PASS: TestSystemConfiguration/correct_configuration_is_returned_for_system_collector (0.00s)
=== RUN   TestSystemCollect
=== RUN   TestSystemCollect/test_default_system_collection
--- PASS: TestSystemCollect (0.24s)
    --- PASS: TestSystemCollect/test_default_system_collection (0.24s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/collectors/system	(cached)
=== RUN   TestGetOutputNames
=== RUN   TestGetOutputNames/can_get_all_output_names
--- PASS: TestGetOutputNames (0.00s)
    --- PASS: TestGetOutputNames/can_get_all_output_names (0.00s)
=== RUN   TestGetOutputByName
=== RUN   TestGetOutputByName/can_get_output_by_name
=== RUN   TestGetOutputByName/fails_on_non_existing_output
--- PASS: TestGetOutputByName (0.00s)
    --- PASS: TestGetOutputByName/can_get_output_by_name (0.00s)
    --- PASS: TestGetOutputByName/fails_on_non_existing_output (0.00s)
=== RUN   TestBuildConfigurationOptions
=== RUN   TestBuildConfigurationOptions/can_get_all_output_configurations
--- PASS: TestBuildConfigurationOptions (0.00s)
    --- PASS: TestBuildConfigurationOptions/can_get_all_output_configurations (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/outputs	(cached)
=== RUN   TestConfiguration
=== RUN   TestConfiguration/correct_configuration_is_returned
--- PASS: TestConfiguration (0.00s)
    --- PASS: TestConfiguration/correct_configuration_is_returned (0.00s)
=== RUN   TestWrite
=== RUN   TestWrite/test_we_can_write_simple_diagnostic_sets
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"creating bundle in /tmp/TestWritetest_we_can_write_simple_diagnostic_sets4152300462/001/test/test"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"bundle created"}
=== RUN   TestWrite/test_invalid_parameter
=== RUN   TestWrite/test_compression
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"creating bundle in /tmp/TestWritetest_compression2256853277/001/test/test"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"bundle created"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"compressing bundle to /tmp/TestWritetest_compression2256853277/001/test/test.tar.gz"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"archive ready at: /tmp/TestWritetest_compression2256853277/001/test/test.tar.gz "}
=== RUN   TestWrite/test_support_for_directory_frames
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"creating bundle in /tmp/TestWritetest_support_for_directory_frames1027574502/002/test/test"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"bundle created"}
=== RUN   TestWrite/test_support_for_config_frames
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"creating bundle in /tmp/TestWritetest_support_for_config_frames4050367424/002/test/test"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"bundle created"}
=== RUN   TestWrite/test_support_for_file_frames
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"creating bundle in /tmp/TestWritetest_support_for_file_frames2017215937/002/test/test"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"bundle created"}
=== RUN   TestWrite/test_support_for_hierarchical_frames
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"creating bundle in /tmp/TestWritetest_support_for_hierarchical_frames3049447326/001/test/test"}
{"level":"info","time":"2022-06-20T21:18:17+02:00","message":"bundle created"}
--- PASS: TestWrite (0.02s)
    --- PASS: TestWrite/test_we_can_write_simple_diagnostic_sets (0.00s)
    --- PASS: TestWrite/test_invalid_parameter (0.00s)
    --- PASS: TestWrite/test_compression (0.01s)
    --- PASS: TestWrite/test_support_for_directory_frames (0.00s)
    --- PASS: TestWrite/test_support_for_config_frames (0.01s)
    --- PASS: TestWrite/test_support_for_file_frames (0.00s)
    --- PASS: TestWrite/test_support_for_hierarchical_frames (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/outputs/file	(cached)
?   	github.com/ClickHouse/clickhouse-diagnostics/internal/outputs/terminal	[no test files]
2022/06/20 21:19:20 Starting container id: 3457f5bd3dd9 image: testcontainers/ryuk:0.3.3
2022/06/20 21:19:20 Waiting for container id 3457f5bd3dd9 image: testcontainers/ryuk:0.3.3
2022/06/20 21:19:20 Container is ready id: 3457f5bd3dd9 image: testcontainers/ryuk:0.3.3
2022/06/20 21:19:20 Starting container id: 11ae24cf4db2 image: clickhouse/clickhouse-server:latest
2022/06/20 21:19:20 Waiting for container id 11ae24cf4db2 image: clickhouse/clickhouse-server:latest
2022/06/20 21:19:21 Container is ready id: 11ae24cf4db2 image: clickhouse/clickhouse-server:latest
=== RUN   TestConnect
=== RUN   TestConnect/can_only_connect_once
--- PASS: TestConnect (0.00s)
    --- PASS: TestConnect/can_only_connect_once (0.00s)
=== RUN   TestGetResourceManager
=== RUN   TestGetResourceManager/get_resource_manager
--- PASS: TestGetResourceManager (0.00s)
    --- PASS: TestGetResourceManager/get_resource_manager (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/platform	(cached)
=== RUN   TestGetConfigParam
=== RUN   TestGetConfigParam/can_find_get_config_param_by_name
=== RUN   TestGetConfigParam/throws_error_on_missing_element
--- PASS: TestGetConfigParam (0.00s)
    --- PASS: TestGetConfigParam/can_find_get_config_param_by_name (0.00s)
    --- PASS: TestGetConfigParam/throws_error_on_missing_element (0.00s)
=== RUN   TestValidateConfig
=== RUN   TestValidateConfig/validate_adds_the_default_and_allows_override
=== RUN   TestValidateConfig/validate_errors_if_missing_param
=== RUN   TestValidateConfig/validate_errors_if_invalid_string_value
=== RUN   TestValidateConfig/allow_empty_string_value_if_specified
=== RUN   TestValidateConfig/validate_errors_if_invalid_string_options_value
=== RUN   TestValidateConfig/allow_empty_string_value_for_StringOptions_if_specified
--- PASS: TestValidateConfig (0.00s)
    --- PASS: TestValidateConfig/validate_adds_the_default_and_allows_override (0.00s)
    --- PASS: TestValidateConfig/validate_errors_if_missing_param (0.00s)
    --- PASS: TestValidateConfig/validate_errors_if_invalid_string_value (0.00s)
    --- PASS: TestValidateConfig/allow_empty_string_value_if_specified (0.00s)
    --- PASS: TestValidateConfig/validate_errors_if_invalid_string_options_value (0.00s)
    --- PASS: TestValidateConfig/allow_empty_string_value_for_StringOptions_if_specified (0.00s)
=== RUN   TestReadStringListValues
=== RUN   TestReadStringListValues/can_find_a_string_list_param
--- PASS: TestReadStringListValues (0.00s)
    --- PASS: TestReadStringListValues/can_find_a_string_list_param (0.00s)
=== RUN   TestReadStringValue
=== RUN   TestReadStringValue/can_find_a_string_param
--- PASS: TestReadStringValue (0.00s)
    --- PASS: TestReadStringValue/can_find_a_string_param (0.00s)
=== RUN   TestReadIntValue
=== RUN   TestReadIntValue/can_find_an_integer_param
--- PASS: TestReadIntValue (0.00s)
    --- PASS: TestReadIntValue/can_find_an_integer_param (0.00s)
=== RUN   TestReadBoolValue
=== RUN   TestReadBoolValue/can_find_a_boolean_param
--- PASS: TestReadBoolValue (0.00s)
    --- PASS: TestReadBoolValue/can_find_a_boolean_param (0.00s)
=== RUN   TestReadStringOptionsValue
=== RUN   TestReadStringOptionsValue/can_find_a_string_value_in_a_list_of_options
=== RUN   TestReadStringOptionsValue/errors_on_invalid_value
--- PASS: TestReadStringOptionsValue (0.00s)
    --- PASS: TestReadStringOptionsValue/can_find_a_string_value_in_a_list_of_options (0.00s)
    --- PASS: TestReadStringOptionsValue/errors_on_invalid_value (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/platform/config	(cached)
=== RUN   TestBundleError
=== RUN   TestBundleError/can_get_a_bundle_error
--- PASS: TestBundleError (0.00s)
    --- PASS: TestBundleError/can_get_a_bundle_error (0.00s)
=== RUN   TestString
=== RUN   TestString/can_order_by_asc
=== RUN   TestString/can_order_by_desc
--- PASS: TestString (0.00s)
    --- PASS: TestString/can_order_by_asc (0.00s)
    --- PASS: TestString/can_order_by_desc (0.00s)
=== RUN   TestNextDatabaseFrame
=== RUN   TestNextDatabaseFrame/can_iterate_sql_rows
=== RUN   TestNextDatabaseFrame/can_iterate_empty_sql_rows
--- PASS: TestNextDatabaseFrame (0.00s)
    --- PASS: TestNextDatabaseFrame/can_iterate_sql_rows (0.00s)
    --- PASS: TestNextDatabaseFrame/can_iterate_empty_sql_rows (0.00s)
=== RUN   TestNextFileDirectoryFrame
=== RUN   TestNextFileDirectoryFrame/can_iterate_file_frame
=== RUN   TestNextFileDirectoryFrame/can_iterate_file_frame_when_empty
--- PASS: TestNextFileDirectoryFrame (0.00s)
    --- PASS: TestNextFileDirectoryFrame/can_iterate_file_frame (0.00s)
    --- PASS: TestNextFileDirectoryFrame/can_iterate_file_frame_when_empty (0.00s)
=== RUN   TestNewConfigFileFrame
=== RUN   TestNewConfigFileFrame/can_iterate_config_file_frame
=== RUN   TestNewConfigFileFrame/can_iterate_file_frame_when_empty
--- PASS: TestNewConfigFileFrame (0.04s)
    --- PASS: TestNewConfigFileFrame/can_iterate_config_file_frame (0.03s)
    --- PASS: TestNewConfigFileFrame/can_iterate_file_frame_when_empty (0.00s)
=== RUN   TestConfigFileFrameCopy
=== RUN   TestConfigFileFrameCopy/can_copy_non-sensitive_xml_config_files
=== RUN   TestConfigFileFrameCopy/can_copy_sensitive_xml_config_files
=== RUN   TestConfigFileFrameCopy/can_copy_sensitive_yaml_config_files
--- PASS: TestConfigFileFrameCopy (0.40s)
    --- PASS: TestConfigFileFrameCopy/can_copy_non-sensitive_xml_config_files (0.03s)
    --- PASS: TestConfigFileFrameCopy/can_copy_sensitive_xml_config_files (0.20s)
    --- PASS: TestConfigFileFrameCopy/can_copy_sensitive_yaml_config_files (0.18s)
=== RUN   TestConfigFileFrameFindLogPaths
=== RUN   TestConfigFileFrameFindLogPaths/can_find_xml_log_paths
=== RUN   TestConfigFileFrameFindLogPaths/can_handle_empty_log_paths
=== RUN   TestConfigFileFrameFindLogPaths/can_find_yaml_log_paths
--- PASS: TestConfigFileFrameFindLogPaths (0.16s)
    --- PASS: TestConfigFileFrameFindLogPaths/can_find_xml_log_paths (0.08s)
    --- PASS: TestConfigFileFrameFindLogPaths/can_handle_empty_log_paths (0.00s)
    --- PASS: TestConfigFileFrameFindLogPaths/can_find_yaml_log_paths (0.08s)
=== RUN   TestYandexConfigFile
=== RUN   TestYandexConfigFile/can_find_xml_log_paths_with_yandex_root
--- PASS: TestYandexConfigFile (0.06s)
    --- PASS: TestYandexConfigFile/can_find_xml_log_paths_with_yandex_root (0.06s)
=== RUN   TestNextMemoryFrame
=== RUN   TestNextMemoryFrame/can_iterate_memory_frame
=== RUN   TestNextMemoryFrame/can_iterate_memory_frame_when_empty
=== RUN   TestNextMemoryFrame/can_iterate_memory_frame_when_empty#01
--- PASS: TestNextMemoryFrame (0.00s)
    --- PASS: TestNextMemoryFrame/can_iterate_memory_frame (0.00s)
    --- PASS: TestNextMemoryFrame/can_iterate_memory_frame_when_empty (0.00s)
    --- PASS: TestNextMemoryFrame/can_iterate_memory_frame_when_empty#01 (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/platform/data	(cached)
2022/06/20 21:19:44 Starting container id: 5ec5c64d4f02 image: testcontainers/ryuk:0.3.3
2022/06/20 21:19:44 Waiting for container id 5ec5c64d4f02 image: testcontainers/ryuk:0.3.3
2022/06/20 21:19:44 Container is ready id: 5ec5c64d4f02 image: testcontainers/ryuk:0.3.3
2022/06/20 21:19:44 Starting container id: 66a84c7e7fed image: clickhouse/clickhouse-server:latest
2022/06/20 21:19:45 Waiting for container id 66a84c7e7fed image: clickhouse/clickhouse-server:latest
2022/06/20 21:19:45 Container is ready id: 66a84c7e7fed image: clickhouse/clickhouse-server:latest
=== RUN   TestReadTableNamesForDatabase
=== RUN   TestReadTableNamesForDatabase/client_can_read_tables_for_a_database
    native_test.go:70: 
        	Error Trace:	native_test.go:70
        	Error:      	Not equal: 
        	            	expected: 70
        	            	actual  : 72
        	Test:       	TestReadTableNamesForDatabase/client_can_read_tables_for_a_database
--- FAIL: TestReadTableNamesForDatabase (0.00s)
    --- FAIL: TestReadTableNamesForDatabase/client_can_read_tables_for_a_database (0.00s)
=== RUN   TestReadTable
=== RUN   TestReadTable/client_can_get_all_rows_for_system.disks_table
=== RUN   TestReadTable/client_can_get_all_rows_for_system.databases_table
=== RUN   TestReadTable/client_can_get_all_rows_for_system.databases_table_with_except
=== RUN   TestReadTable/client_can_limit_rows_for_system.databases
=== RUN   TestReadTable/client_can_order_rows_for_system.databases
--- PASS: TestReadTable (0.01s)
    --- PASS: TestReadTable/client_can_get_all_rows_for_system.disks_table (0.00s)
    --- PASS: TestReadTable/client_can_get_all_rows_for_system.databases_table (0.00s)
    --- PASS: TestReadTable/client_can_get_all_rows_for_system.databases_table_with_except (0.00s)
    --- PASS: TestReadTable/client_can_limit_rows_for_system.databases (0.00s)
    --- PASS: TestReadTable/client_can_order_rows_for_system.databases (0.00s)
=== RUN   TestExecuteStatement
=== RUN   TestExecuteStatement/client_can_execute_any_statement
1
--- PASS: TestExecuteStatement (0.00s)
    --- PASS: TestExecuteStatement/client_can_execute_any_statement (0.00s)
=== RUN   TestVersion
=== RUN   TestVersion/client_can_read_version
--- PASS: TestVersion (0.00s)
    --- PASS: TestVersion/client_can_read_version (0.00s)
FAIL
FAIL	github.com/ClickHouse/clickhouse-diagnostics/internal/platform/database	1.354s
2022/06/20 21:18:18 Starting container id: 0a326be31edf image: testcontainers/ryuk:0.3.3
2022/06/20 21:18:18 Waiting for container id 0a326be31edf image: testcontainers/ryuk:0.3.3
2022/06/20 21:18:18 Container is ready id: 0a326be31edf image: testcontainers/ryuk:0.3.3
2022/06/20 21:18:18 Starting container id: 0bd4c6bcbede image: clickhouse/clickhouse-server:latest
2022/06/20 21:18:19 Waiting for container id 0bd4c6bcbede image: clickhouse/clickhouse-server:latest
2022/06/20 21:18:20 Container is ready id: 0bd4c6bcbede image: clickhouse/clickhouse-server:latest
=== RUN   TestFileExists
=== RUN   TestFileExists/returns_true_for_file
=== RUN   TestFileExists/doesn't_return_true_for_not_existence_file
=== RUN   TestFileExists/doesn't_return_true_for_directory
--- PASS: TestFileExists (0.00s)
    --- PASS: TestFileExists/returns_true_for_file (0.00s)
    --- PASS: TestFileExists/doesn't_return_true_for_not_existence_file (0.00s)
    --- PASS: TestFileExists/doesn't_return_true_for_directory (0.00s)
=== RUN   TestDirExists
=== RUN   TestDirExists/doesn't_return_true_for_file
=== RUN   TestDirExists/returns_true_for_directory
=== RUN   TestDirExists/doesn't_return_true_random_directory
--- PASS: TestDirExists (0.00s)
    --- PASS: TestDirExists/doesn't_return_true_for_file (0.00s)
    --- PASS: TestDirExists/returns_true_for_directory (0.00s)
    --- PASS: TestDirExists/doesn't_return_true_random_directory (0.00s)
=== RUN   TestCopyFile
=== RUN   TestCopyFile/can_copy_file
=== RUN   TestCopyFile/can_copy_nested_file
=== RUN   TestCopyFile/fails_when_file_does_not_exist
--- PASS: TestCopyFile (0.00s)
    --- PASS: TestCopyFile/can_copy_file (0.00s)
    --- PASS: TestCopyFile/can_copy_nested_file (0.00s)
    --- PASS: TestCopyFile/fails_when_file_does_not_exist (0.00s)
=== RUN   TestListFilesInDirectory
=== RUN   TestListFilesInDirectory/can_list_all_files
=== RUN   TestListFilesInDirectory/can_list_by_extension
=== RUN   TestListFilesInDirectory/can_list_on_multiple_extensions_files
--- PASS: TestListFilesInDirectory (0.00s)
    --- PASS: TestListFilesInDirectory/can_list_all_files (0.00s)
    --- PASS: TestListFilesInDirectory/can_list_by_extension (0.00s)
    --- PASS: TestListFilesInDirectory/can_list_on_multiple_extensions_files (0.00s)
=== RUN   TestFindClickHouseProcesses
=== RUN   TestFindClickHouseProcesses/can_find_ClickHouse_processes
--- PASS: TestFindClickHouseProcesses (0.01s)
    --- PASS: TestFindClickHouseProcesses/can_find_ClickHouse_processes (0.01s)
=== RUN   TestFindConfigsFromClickHouseProcesses
=== RUN   TestFindConfigsFromClickHouseProcesses/can_find_ClickHouse_configs
--- PASS: TestFindConfigsFromClickHouseProcesses (0.01s)
    --- PASS: TestFindConfigsFromClickHouseProcesses/can_find_ClickHouse_configs (0.01s)
=== RUN   TestIntersection
=== RUN   TestIntersection/can_perform_intersection
--- PASS: TestIntersection (0.00s)
    --- PASS: TestIntersection/can_perform_intersection (0.00s)
=== RUN   TestDistinct
=== RUN   TestDistinct/can_perform_distinct
=== RUN   TestDistinct/can_perform_distinct_on_empty
--- PASS: TestDistinct (0.00s)
    --- PASS: TestDistinct/can_perform_distinct (0.00s)
    --- PASS: TestDistinct/can_perform_distinct_on_empty (0.00s)
=== RUN   TestContains
=== RUN   TestContains/can_perform_contains
--- PASS: TestContains (0.00s)
    --- PASS: TestContains/can_perform_contains (0.00s)
=== RUN   TestUnique
=== RUN   TestUnique/can_perform_unique
=== RUN   TestUnique/can_perform_unique_on_empty
--- PASS: TestUnique (0.00s)
    --- PASS: TestUnique/can_perform_unique (0.00s)
    --- PASS: TestUnique/can_perform_unique_on_empty (0.00s)
PASS
ok  	github.com/ClickHouse/clickhouse-diagnostics/internal/platform/utils	(cached)
FAIL
```

I've updated `clickhouse/clickhouse-server:latest`

--- a/tools/clickhouse-diagnostics/internal/platform/database/native_test.go
+++ b/tools/clickhouse-diagnostics/internal/platform/database/native_test.go
@@ -67,7 +67,7 @@ func TestReadTableNamesForDatabase(t *testing.T) {
 	t.Run("client can read tables for a database", func(t *testing.T) {
 		tables, err := clickhouseClient.ReadTableNamesForDatabase("system")
 		require.Nil(t, err)
-		require.Greater(t, 70, len(tables))
+		require.GreaterOrEqual(t, len(tables), 70)
 		require.Contains(t, tables, "merge_tree_settings")
 	})
 }

--- a/tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.err.log
+++ b/tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.err.log
@@ -1,0 +1,10 @@
+2021.12.13 10:12:26.940169 [ 38398 ] {} <Warning> Access(local directory): File /var/lib/clickhouse/access/users.list doesn't exist
+2021.12.13 10:12:26.940204 [ 38398 ] {} <Warning> Access(local directory): Recovering lists in directory /var/lib/clickhouse/access/
+2021.12.13 10:12:40.649453 [ 38445 ] {} <Error> Access(user directories): from: 127.0.0.1, user: default: Authentication failed: Code: 193. DB::Exception: Invalid credentials. (WRONG_PASSWORD), Stack trace (when copying this message, always include the lines below):
+
+0. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0x9b722d4 in /usr/bin/clickhouse
+1. DB::IAccessStorage::throwInvalidCredentials() @ 0x119d9b27 in /usr/bin/clickhouse
+2. DB::IAccessStorage::loginImpl(DB::Credentials const&, Poco::Net::IPAddress const&, DB::ExternalAuthenticators const&) const @ 0x119d98d7 in /usr/bin/clickhouse
+3. DB::IAccessStorage::login(DB::Credentials const&, Poco::Net::IPAddress const&, DB::ExternalAuthenticators const&, bool) const @ 0x119d9084 in /usr/bin/clickhouse
+4. DB::MultipleAccessStorage::loginImpl(DB::Credentials const&, Poco::Net::IPAddress const&, DB::ExternalAuthenticators const&) const @ 0x119ff93c in /usr/bin/clickhouse
+5. DB::IAccessStorage::login(DB::Credentials const&, Poco::Net::IPAddress const&, DB::ExternalAuthenticators const&, bool) const @ 0x119d9084 in /usr/bin/clickhouse

--- a/tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.log
+++ b/tools/clickhouse-diagnostics/testdata/logs/var/logs/clickhouse-server.log
@@ -1,0 +1,10 @@
+2022.02.02 14:49:32.458680 [ 200404 ] {} <Debug> DiskLocal: Reserving 2.47 MiB on disk `default`, having unreserved 1.56 TiB.
+2022.02.02 14:49:32.459086 [ 200359 ] {de87df8b-2250-439c-9e87-df8b2250339c::202202_147058_147550_344} <Debug> MergeTask::PrepareStage: Merging 2 parts: from 202202_147058_147549_343 to 202202_147550_147550_0 into Wide
+2022.02.02 14:49:32.459201 [ 200359 ] {de87df8b-2250-439c-9e87-df8b2250339c::202202_147058_147550_344} <Debug> MergeTask::PrepareStage: Selected MergeAlgorithm: Horizontal
+2022.02.02 14:49:32.459262 [ 200359 ] {de87df8b-2250-439c-9e87-df8b2250339c::202202_147058_147550_344} <Debug> MergeTreeSequentialSource: Reading 159 marks from part 202202_147058_147549_343, total 1289014 rows starting from the beginning of the part
+2022.02.02 14:49:32.459614 [ 200359 ] {de87df8b-2250-439c-9e87-df8b2250339c::202202_147058_147550_344} <Debug> MergeTreeSequentialSource: Reading 2 marks from part 202202_147550_147550_0, total 2618 rows starting from the beginning of the part
+2022.02.02 14:49:32.507755 [ 200359 ] {de87df8b-2250-439c-9e87-df8b2250339c::202202_147058_147550_344} <Debug> MergeTask::MergeProjectionsStage: Merge sorted 1291632 rows, containing 5 columns (5 merged, 0 gathered) in 0.048711404 sec., 26516008.448452853 rows/sec., 639.52 MiB/sec.
+2022.02.02 14:49:32.508332 [ 200359 ] {de87df8b-2250-439c-9e87-df8b2250339c::202202_147058_147550_344} <Trace> system.asynchronous_metric_log (de87df8b-2250-439c-9e87-df8b2250339c): Renaming temporary part tmp_merge_202202_147058_147550_344 to 202202_147058_147550_344.
+2022.02.02 14:49:32.508406 [ 200359 ] {de87df8b-2250-439c-9e87-df8b2250339c::202202_147058_147550_344} <Trace> system.asynchronous_metric_log (de87df8b-2250-439c-9e87-df8b2250339c) (MergerMutator): Merged 2 parts: from 202202_147058_147549_343 to 202202_147550_147550_0
+2022.02.02 14:49:32.508440 [ 200359 ] {} <Debug> MemoryTracker: Peak memory usage Mutate/Merge: 16.31 MiB.
+2022.02.02 14:49:33.000148 [ 200388 ] {} <Trace> AsynchronousMetrics: MemoryTracking: was 774.16 MiB, peak 2.51 GiB, will set to 772.30 MiB (RSS), difference: -1.86 MiB

